woocommerce/woocommerce-gutenberg-products-block
# Create Reviews by Product block

Fixes #588.

**Note:** don't merge this PR to master until all TODO follow-ups are merged into this branch.



#### Accessibility




### Screenshots

### How to test the changes in this Pull Request:

1. In a new or existing post/page add a _Reviews by Product_ block.
2. Try interacting with the settings on the sidebar and verify the block updates accordingly.



### Todo
* [x] Refactor styles so the block is styled in the editor.
* [x] Update sorting options to _Most Recent_, _Highest Rating_ and _Lowest Rating_.
* [x] Add icon.
* [x] Allow additional CSS classes attribute.
* [x] Show Reviews count next to the product name in the Product Control (#671).
* [x] Reviews count should display X Reviews instead of only the number. That requires updating the WooCommerce Admin component (#677).
![image](https://user-images.githubusercontent.com/3616980/60584515-c1ddd100-9d8d-11e9-9805-90861aa1d8ec.png)
* [x] Create placeholder and make the block save it in the markup (#691). @Aljullu 
* [x] Render reviews in JavaScript instead of PHP (#696). @Aljullu
* [x] Add _Order by_ select to the frontend component (#716). @Aljullu 
* [x] Add  _Load more_ button to the frontend component (#716). @Aljullu 
* [x] Hide products with no reviews from the Product Control (#688). @mikejolley
* [x] Add placeholder to Product Control when no products have reviews. @mikejolley
* [x] Rename _Reviewer Picture_ with _Avatar_ (#702). @Aljullu 
* [x] Extend reviews API with a blocks endpoint. @mikejolley 
* [x] Hookup the _Order by_ options with the new endpoint (#736). @Aljullu 
* [x] IE11 testing and style-fixing (#723). @Aljullu 
* [x] Hide avatars if `get_option( 'show_avatars' )` is false (#730). @Aljullu
* [x] Hide ratings if they are disabled in WooCommerce Settings. @Aljullu 
* [x] Reduce the number of dependencies required in the frontend (#774). @Aljullu 
* [x] Create a `withProduct()` HOC so fetching logic can be shared with _Featured Product_ (#779). @Aljullu 
* [x] Use the `withProduct()` HOC implemented in the previous issue (#828). @Aljullu 
* [x] Add _Display Order by_ setting (#787). @Aljullu 
* [x] Rename _Reviews shown on page load_ setting to _Starting Number of Reviews_ and use a slider (#787). @Aljullu 
* [x] Add _Display Load More_ setting (#787). @Aljullu 
* [x] Add _Load More Reviews_ setting using a slider (#787). @Aljullu 
* [x] Add _Image Type_ setting (#787). @Aljullu 
* [x] Trim long reviews and add a _Read full review_ button. @mikejolley
* [x] Add _Verified_ icon next to reviewer name if they have purchased the product in the past. @mikejolley
* [x] Update reviews design according to last mock-ups. @mikejolley
